lib/neurx/structures/layer.ex

```diff
@@ -22,13 +24,15 @@ defmodule Neurx.Layer do
   @doc """
   Return a layer by pid.
   """
-  def get(pid), do: Agent.get(pid, & &1)
-
+  def get(pid) do
+    GenServer.call(pid, {:get})
+  end
   @doc """
   Update a layer by passing in a pid and a map of fields to update.
   """
   def update(pid, fields) do
-    Agent.update(pid, &Map.merge(&1, fields))
+    # why dont preserve pid
```

Comment:
Unnecessary comment. PIDs are not preserved since they can change if a process needs to be restarted.

Reply:
I forget to put it but the all the test have passed so i didnt notice it